### PR TITLE
Update network example notebook

### PR DESCRIPTION
**Pull request by dwhswenson**
This brings this notebook in line with openfe 0.4 and gufe 0.4. Also adds it to testing.

### CODE REVIEW

File: networks/ligand_networks_for_developers.ipynb [hunk 1]
```diff
@@ -138,17 +137,16 @@
...
    "source": [
-    "from openfe.setup import LigandAtomMapper\n",
+    "from openfe.setup.atom_mapping import LigandAtomMapper\n",
     "\n",
     "class CustomMapper(LigandAtomMapper):\n",
     "    def __init__(self, remove_middle):\n",
     "        super().__init__()\n",
     "        self.remove_middle = remove_middle\n",
     "    \n",
     "    def _mappings_generator(self, molA, molB):\n",
```
Comment (IAlibay) [inline networks/ligand_networks_for_developers.ipynb:147]: Do we want to rename things to `componentA/B` or is it suitably clear if we keep `molA/B`?

File: networks/ligand_networks_for_developers.ipynb [hunk 2]
```diff
@@ -646,11 +644,11 @@
     "    # This score is the highest indices of mapped atoms plus the length of the molecule names\n",
     "    # It shows you how to access this information from the mapping. I wouldn't recommend\n",
     "    # it for production work.\n",
-    "    molA_to_molB = mapping.molA_to_molB\n",
+    "    molA_to_molB = mapping.componentA_to_componentB\n",
```
Comment (IAlibay) [inline networks/ligand_networks_for_developers.ipynb:647]: One option might just be to explain why it's called `component` somewhere?

**Review by IAlibay — APPROVED** [review]
Just the one question

File: networks/ligand_networks_for_developers.ipynb (hunk 1 above)
Comment (richardjgowers) [reply networks/ligand_networks_for_developers.ipynb:147]: I think it's probably better of two evils to use componentA/B consistently and explain why.

Comment (dwhswenson) [reply networks/ligand_networks_for_developers.ipynb:147]: Done! please revisit. I didn't completely excise the word "molecule," but I used component where I thought it made sense, and added a bit about components.

**Comment by dwhswenson** [comment]
wait.. my push didn't take

**Comment by dwhswenson** [comment]
ok, there it is... Working from a outside my home, looks like WiFi here blocked SSH activity. Just used my phone as a hotspot.

**Review by IAlibay — APPROVED** [review]

**Pull request merged**